src/Database/Type/BoolType.php

```diff
@@ -24,10 +24,40 @@
  *
  * Use to convert bool data between PHP and the database types.
  */
-class BoolType extends Type
+class BoolType implements TypeInterface
 {
 
     /**
+     * Identifier name for this type
+     *
+     * @var string
+     */
+    protected $_name = null;
+
+    /**
+     * Constructor
+     *
+     * @param string|null $name The name identifying this type
+     */
+    public function __construct($name = null)
```

Comment:
It's duplicated in `binaryType`, `boolType`, `floatType` and `intergerType`.

I had two thoughts, the first was a trait, and the second was adding it to the interface. A trait would work, but on reflection I thought that perhaps there was a software design issue here which could be solved. Adding it to the interface isn't a good idea, as it's not consistently needed across all the type classes.

Glad you spotted it too, I'd be interested in getting other peoples thoughts on it.